exercises/alphametics/HINTS.md

```diff
@@ -0,0 +1,25 @@
+## Hints
+
+You can solve this exercise with a brute force algorithm, but this will possibly have a poor runtime performance.
+Try to find a more sophisticated solution. Hint: You could try the column-wise addition algorithm that is usually taught in high school.
+
+Once all the tests succeed you can go further.
+For example, your solution might still run into trouble with more characters:
+```haskell
+        , Case { description  = "puzzle with ten letters"
+               , puzzle       = "AND + A + STRONG + OFFENSE + AS + A + GOOD == DEFENSE"
+               , expected     = Just [('A', 5), ('D', 3), ('E', 4), ('F', 7), ('G', 8),
+                                      ('N', 0), ('O', 2), ('R', 1), ('S', 6), ('T', 9)]
+               }
+```
+Or you can try puzzles with other operations like multiplication and exponentiation:
+```haskell
+        , Case { description  = "puzzle with multiplication"
+               , puzzle       = "IF * DR == DORI"
+               , expected     = Just [('I', 8), ('F', 2), ('D', 3), ('R', 9), ('O', 1)]
+               }
+        , Case { description  = "puzzle with exponents"
+               , puzzle       = "PI * R ^ 2 == AREA"
+               , expected     = Just [('P', 9), ('I', 6), ('R', 7), ('A', 4), ('E', 0)]
+               }
+```
```

Comment:
I feel that these cases pollute the `HINTS.md` file. Also, I don't believe anyone will copy/paste this in `test/Test.hs`.

I prefer them commented out in the test suite, but I will respect your decision if you prefer them here, @abo64.